.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+on:
+  push:
+  pull_request:
+
+jobs:
+  generate-winmd:
+    name: Generate winmd
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Clean
+        run: Remove-Item -Recurse -Force .windows/winmd/
+      - uses: ilammy/msvc-dev-cmd@v1 # Needed to find CL.EXE
+      - name: Generate
+        run:  dotnet build
+        working-directory: .metadata/
+      - name: Upload
+        uses: actions/upload-artifact@v3
+        with:
+          name: metadata
+          # TODO: Only need .windows/winmd/Microsoft.States.winmd,
+          # the core Win32 bindings are 15MB and don't seem needed
+          # to generate the Rust code since windows-metadata already
+          # includes these files.
```

Comment:
@riverar Not super sure about this, it seems the `windows-metadata` Rust crate already ships these files but they may get out of sync with `WinmdGenerator` where they are copied from?

It seems the generation is mostly fine without them because no `Win32` types are used anywhere; even without `Win32_Foundation` feature it all builds and runs fine (and no use of this namespace is observed anywhere).

Reply:
I just sourced these winmds from the `windows-rs` metadata default folder manually. Yep, they can definitely get out of sync with `windows-metadata` and/or `windows-rs`. There's no good fix at this time due to cargo deficiencies. So a release policy will have to fill that gap for now.

Reply:
> I just sourced these winmds from the `windows-rs` metadata default folder manually.

Aren't they copied from the `WinmdGenerator` (or a subpackage of that)?

https://github.com/riverar/washington-rs/blob/dc6de8a6ffbf815c9c4fda26f3e5d35cbe8f5d21/.metadata/generate.proj#L19-L27

---

I thought I later - maybe with https://github.com/microsoft/DirectX-Headers/pull/82 - saw that relying on `windows-rs` bundled `.winmd` either didn't work or was out of sync with the `.winmd` generated with a more recent `WinmdGenerator`, so it is perhaps best to distribute these with it? Or strip it out altogether?

(As mentioned above `washington-rs` might be fine because it didn't import external types)

Reply:
Oh I forgot I wired this up to the metadata Nuget package! Yeah it's only there to show off that functionality for a few teams that emailed me. It's not needed (yet) by the crate. We can pretend it's needed or you can delete it. I have no preference either way.

Reply:
Mostly so curious if the files are really needed since they're super large compared to the generated `.winmd`, but I think I've answered my own question above (it doesn't work otherwise). Happy to keep it around, but would like to document (via some in-line comments) their existence to that effect.